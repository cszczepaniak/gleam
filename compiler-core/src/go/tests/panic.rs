use crate::assert_go;

#[test]
fn bare() {
    assert_go!(
        r#"
fn go() {
  panic
}
"#,
    );
}
//
// #[test]
// fn panic_as() {
//     assert_js!(
//         r#"
// fn go() {
//   let x = "wibble"
//   panic as x
// }
// "#,
//     );
// }
//
// #[test]
// fn bare_typescript() {
//     assert_ts_def!(
//         r#"
// pub fn go() {
//   panic
// }
// "#,
//     );
// }
//
// #[test]
// fn as_expression() {
//     assert_js!(
//         r#"
// fn go(f) {
//   let boop = panic
//   f(panic)
// }
// "#,
//     );
// }
//
// #[test]
// fn pipe() {
//     assert_js!(
//         r#"
// fn go(f) {
//   f |> panic
// }
// "#,
//     );
// }
//
// #[test]
// fn sequence() {
//     assert_js!(
//         r#"
// fn go(at_the_disco) {
//   panic
//   at_the_disco
// }
// "#,
//     );
// }
//
// #[test]
// fn case() {
//     assert_js!(
//         r#"
// fn go(x) {
//   case x {
//     _ -> panic
//   }
// }
// "#,
//     );
// }
