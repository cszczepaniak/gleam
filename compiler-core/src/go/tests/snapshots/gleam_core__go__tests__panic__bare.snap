---
source: compiler-core/src/go/tests/panic.rs
expression: "\nfn a() {\n  panic\n}\n"
---
package mod

func a() {
  panic("panic expression evaluated")
}
